cargo.py: default to 0.0.0 if version field is missing

jpnssl opened the pull request:
Starting with Rust 1.75, the [version] field in cargo manifests is no longer mandatory, defaulting to 0.0.0 if unset.(1)

Fixes: #42

(1) https://doc.rust-lang.org/cargo/reference/manifest.html#the-version-field

mgorny commented:
Thanks. Could you:

1. Add a signoff per Gentoo copyright policy,
2. Update tests to cover this case,
3. Add a comment linking to cargo book above the fallback,
4. Update the commit message to link bugs using the full URL?

jpnssl committed:
cargo: default to 0.0.0 if version field is missing

Starting with Rust 1.75, the [version] field in cargo manifests is no longer mandatory, defaulting to 0.0.0 if unset.(1)

Fixes: https://github.com/gentoo/pycargoebuild/issues/42

(1) https://doc.rust-lang.org/cargo/reference/manifest.html#the-version-field

Signed-off-by: Johannes Penßel <johannesp@posteo.net>

mgorny reviewed (approved):
Thanks. Lemme test, and then release.
